tools/mo/openvino/tools/mo/ops/gather.py

```diff
@@ -100,6 +101,38 @@ def infer(node: Node):
         else:
             node.out_port(0).data.set_shape(out_shape)
 
+    @staticmethod
+    def reverse_infer(node: Node):
+        out_shape = node.out_port(0).data.get_shape()
+        data_shape = node.in_port(0).data.get_shape()
+        indices_shape = node.in_port(1).data.get_shape()
+        batch_dims = node.batch_dims
+        batch_dims = batch_dims + len(indices_shape) if batch_dims < 0 else batch_dims
+        axis = node.in_port(2).data.get_value()
+
```

Comment:
please check that axis is not None
also, please extract scalar axis.
```
        if isinstance(axis, np.ndarray):
            axis = axis.item()
        assert axis is not None, 'axis input is undefined'
```

Reply:
done